tools/wptrunner/wptrunner/executors/base.py

```diff
@@ -620,8 +621,23 @@ def retake_screenshot(self, node, viewport_size, dpi, page_ranges):
 
     def get_screenshot_list(self, node, viewport_size, dpi, page_ranges):
         success, data = self.executor.screenshot(node, viewport_size, dpi, page_ranges)
-        if success and not isinstance(data, list):
-            return success, [data]
+        viewport_size = (800, 600) if viewport_size is None else viewport_size
+
+        if not isinstance(data, list):
+            data = [data]
+
+        if success:
+            for screenshot in data:
+                image_data = base64.b64decode(screenshot)
```

Comment:
nit: Maybe just decode the bytes we need? If my math is right:

```suggestion
                image_data = base64.b64decode(screenshot[:32])
```